examples/run/cpu/seq_batch_example.cpp

```diff
@@ -0,0 +1,147 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021-2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// io
+#include "traccc/io/read.hpp"
+
+// algorithms
+#include "traccc/clusterization/clusterization_algorithm.hpp"
+#include "traccc/clusterization/spacepoint_formation.hpp"
+#include "traccc/seeding/seeding_algorithm.hpp"
+#include "traccc/seeding/track_params_estimation.hpp"
+
+// performance
+#include "traccc/performance/timer.hpp"
+
+// options
+#include "traccc/options/batch_full_tracking_input_options.hpp"
+#include "traccc/options/common_options.hpp"
+#include "traccc/options/handle_argument_errors.hpp"
+
+// VecMem include(s).
+#include <vecmem/memory/host_memory_resource.hpp>
+
+// System include(s).
+#include <time.h>
+
+#include <cstdlib>
+#include <exception>
+#include <iostream>
+
+namespace po = boost::program_options;
+
+int seq_batch_run(const traccc::batch_full_tracking_input_config& i_cfg,
+                  const traccc::common_options& common_opts) {
+
+    // Memory resource used by the EDM.
+    vecmem::host_memory_resource host_mr;
+
+    traccc::clusterization_algorithm ca(host_mr);
+    traccc::spacepoint_formation sf(host_mr);
+    traccc::seeding_algorithm sa(host_mr);
+    traccc::track_params_estimation tp(host_mr);
+
+    traccc::demonstrator_input cells_event_vec;
+    cells_event_vec.reserve(common_opts.events);
+
+    traccc::performance::timing_info elapsedTimes;
+
+    {
+
+        traccc::performance::timer t("File reading", elapsedTimes);
+
+        cells_event_vec = traccc::io::read(
+            common_opts.events, common_opts.input_directory,
+            i_cfg.detector_file, i_cfg.digitization_config_file,
+            common_opts.input_data_format, &host_mr);
+    }
+    // Seed random number generator
+    std::srand(std::time(0));
+
+    // Loop over events
+    for (unsigned int i = 0; i < i_cfg.batch_events; ++i) {
+
+        const unsigned int event = rand() % common_opts.events;
```

Comment:
For consistency you should use the `std::rand()` name here. Other than that, this type of simple randomisation is perfect for such applications.